### PR TITLE
[auth](enhance)remove some method in AccessControllerManager

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

The authentication method must specify the catalog level to avoid forgetting to pass the catalogName

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CleanQueryStatsStmt.java [hunk 1]
```diff
@@ -95,7 +96,8 @@ public void analyze(Analyzer analyzer) throws UserException {
 
                 Env.getCurrentEnv().getCurrentCatalog().getDbOrAnalysisException(dbName);
                 if (!Env.getCurrentEnv().getAccessManager()
-                        .checkDbPriv(ConnectContext.get(), dbName, PrivPredicate.ALTER)) {
+                        .checkDbPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME, dbName,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CleanQueryStatsStmt.java:99]: No external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DropDbStmt.java [hunk 2]
```diff
@@ -73,9 +74,10 @@ public void analyze(Analyzer analyzer) throws UserException {
                     analyzer.getQualifiedUser(), dbName);
         }
 
-        if (!Env.getCurrentEnv().getAccessManager().checkDbPriv(ConnectContext.get(), dbName, PrivPredicate.DROP)) {
+        if (!Env.getCurrentEnv().getAccessManager()
+                .checkDbPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME, dbName, PrivPredicate.DROP)) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DropDbStmt.java:78]: External catalog

File: fe/fe-core/src/main/java/org/apache/doris/analysis/DropStatsStmt.java [hunk 3]
```diff
@@ -175,7 +175,8 @@ public String toString() {
 
     private void checkAnalyzePriv(String dbName, String tblName) throws AnalysisException {
         if (!Env.getCurrentEnv().getAccessManager()
-                .checkTblPriv(ConnectContext.get(), dbName, tblName, PrivPredicate.DROP)) {
+                .checkTblPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME, dbName, tblName,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/DropStatsStmt.java:178]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowAnalyzeStmt.java [hunk 4]
```diff
@@ -144,7 +145,8 @@ public RedirectStatus getRedirectStatus() {
 
     private void checkShowAnalyzePriv(String dbName, String tblName) throws AnalysisException {
         if (!Env.getCurrentEnv().getAccessManager()
-                .checkTblPriv(ConnectContext.get(), dbName, tblName, PrivPredicate.SHOW)) {
+                .checkTblPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME, dbName, tblName,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowAnalyzeStmt.java:148]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowCreateDbStmt.java [hunk 5]
```diff
@@ -57,8 +58,9 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
             ErrorReport.reportAnalysisException(ErrorCode.ERR_WRONG_DB_NAME, db);
         }
 
-        if (!Env.getCurrentEnv().getAccessManager().checkDbPriv(ConnectContext.get(), db,
-                PrivPredicate.ALTER_CREATE_DROP)) {
+        if (!Env.getCurrentEnv().getAccessManager()
+                .checkDbPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME, db,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowCreateDbStmt.java:62]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowTypeCastStmt.java [hunk 6]
```diff
@@ -70,7 +71,8 @@ public void analyze(Analyzer analyzer) throws UserException {
             }
         }
 
-        if (!Env.getCurrentEnv().getAccessManager().checkDbPriv(ConnectContext.get(), dbName, PrivPredicate.SHOW)) {
+        if (!Env.getCurrentEnv().getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowTypeCastStmt.java:74]: Remove `db` of this stmt, it is useless and no need to check priv for this stmt

File: fe/fe-core/src/main/java/org/apache/doris/cloud/analysis/UseCloudClusterStmt.java [hunk 7]
```diff
@@ -100,7 +101,9 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
         if (Strings.isNullOrEmpty(database)) {
             return;
         }
-        if (!Env.getCurrentEnv().getAccessManager().checkDbPriv(ConnectContext.get(), database, PrivPredicate.SHOW)) {
+        if (!Env.getCurrentEnv().getAccessManager()
+                .checkDbPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME, database,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/analysis/UseCloudClusterStmt.java:105]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/controller/BaseController.java [hunk 8]
```diff
@@ -227,15 +228,17 @@ protected void checkGlobalAuth(UserIdentity currentUser, PrivPredicate predicate
 
     protected void checkDbAuth(UserIdentity currentUser, String db, PrivPredicate predicate)
             throws UnauthorizedException {
-        if (!Env.getCurrentEnv().getAccessManager().checkDbPriv(currentUser, db, predicate)) {
+        if (!Env.getCurrentEnv().getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/controller/BaseController.java:231]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertJob.java [hunk 9]
```diff
@@ -556,15 +557,17 @@ private void checkAuthWithoutAuthInfo(String command) throws DdlException {
         // check auth
         if (tableNames == null || tableNames.isEmpty()) {
             // forward compatibility
-            if (!Env.getCurrentEnv().getAccessManager().checkDbPriv(ConnectContext.get(), db.getFullName(),
-                    PrivPredicate.LOAD)) {
+            if (!Env.getCurrentEnv().getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertJob.java:560]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/BatchInsertIntoTableCommand.java [hunk 10]
```diff
@@ -131,7 +132,8 @@ public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
             }
             // check auth
             if (!Env.getCurrentEnv().getAccessManager()
-                    .checkTblPriv(ConnectContext.get(), targetTable.getQualifiedDbName(), targetTable.getName(),
+                    .checkTblPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/BatchInsertIntoTableCommand.java:135]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertOverwriteTableCommand.java [hunk 11]
```diff
@@ -131,7 +132,8 @@ public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
                     .checkDatabase(((OlapTable) targetTable).getQualifiedDbName(), ConnectContext.get());
             // check auth
             if (!Env.getCurrentEnv().getAccessManager()
-                    .checkTblPriv(ConnectContext.get(), ((OlapTable) targetTable).getQualifiedDbName(),
+                    .checkTblPriv(ConnectContext.get(), InternalCatalog.INTERNAL_CATALOG_NAME,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertOverwriteTableCommand.java:135]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 12]
```diff
@@ -452,7 +452,8 @@ public TGetDbsResult getDbNames(TGetDbsParams params) throws TException {
             }
             for (DatabaseIf db : dbs) {
                 String dbName = db.getFullName();
-                if (!env.getAccessManager().checkDbPriv(currentUser, dbName, PrivPredicate.SHOW)) {
+                if (!env.getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:455]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 13]
```diff
@@ -537,7 +538,8 @@ public TGetTablesResult getTableNames(TGetTablesParams params) throws TException
                         LOG.debug("get table: {}, wait to check", tableName);
                     }
                     if (!Env.getCurrentEnv().getAccessManager()
-                            .checkTblPriv(currentUser, dbName, tableName, PrivPredicate.SHOW)) {
+                            .checkTblPriv(currentUser, InternalCatalog.INTERNAL_CATALOG_NAME, dbName, tableName,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:541]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 14]
```diff
@@ -601,8 +603,9 @@ public TListTableStatusResult listTableStatus(TGetTablesParams params) throws TE
                         }
                     }
                     for (TableIf table : tables) {
-                        if (!Env.getCurrentEnv().getAccessManager().checkTblPriv(currentUser, dbName,
-                                table.getName(), PrivPredicate.SHOW)) {
+                        if (!Env.getCurrentEnv().getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:606]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 15]
```diff
@@ -1083,7 +1086,8 @@ private void checkPasswordAndPrivs(String user, String passwd, String db, List<S
 
         Preconditions.checkState(currentUser.size() == 1);
         if (tables == null || tables.isEmpty()) {
-            if (!Env.getCurrentEnv().getAccessManager().checkDbPriv(currentUser.get(0), fullDbName, predicate)) {
+            if (!Env.getCurrentEnv().getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:1089]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 16]
```diff
@@ -1092,7 +1096,9 @@ private void checkPasswordAndPrivs(String user, String passwd, String db, List<S
         }
 
         for (String tbl : tables) {
-            if (!Env.getCurrentEnv().getAccessManager().checkTblPriv(currentUser.get(0), fullDbName, tbl, predicate)) {
+            if (!Env.getCurrentEnv().getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:1099]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 17]
```diff
@@ -2317,21 +2323,24 @@ public TCheckAuthResult checkAuth(TCheckAuthRequest request) throws TException {
             }
         } else if (privHier == TPrivilegeHier.DATABASE) {
             String fullDbName = privCtrl.getDb();
-            if (!accessManager.checkDbPriv(currentUser.get(0), fullDbName, predicate)) {
+            if (!accessManager.checkDbPriv(currentUser.get(0), InternalCatalog.INTERNAL_CATALOG_NAME, fullDbName,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:2326]: external catalog?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowCreateDbStmt.java (hunk 5 above)
Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/ShowCreateDbStmt.java:62]: Unable to obtain catalog name

File: fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertJob.java (hunk 9 above)
Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertJob.java:560]: Unable to obtain catalog name

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java (hunk 16 above)
Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:1099]: Unable to obtain catalog name

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java (hunk 15 above)
Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:1089]: Unable to obtain catalog name

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by BePPPower — APPROVED** [review]

**Pull request merged**